### PR TITLE
Split up amp embed cleaner

### PR DESCRIPTION
**Pull request by michaelwmcnamara**
## What does this change?
This changes the Amp Embed Cleaner to both make it more readable, and allow elements that have been embedded as a generic "element-embed" figure to be identified and converted to valid amp embeds.
Also changed, more tests are added to the Amp Embed Cleaner tests, to confirm that generic "element-embed" figures will convert to valid amp embeds.

This effects Amp pages. There should be no impact elsewhere.

## Tested in CODE?
Not yet

amp test suite passes
article test suite passes
make validate amp passes

@guardian/dotcom-platform

### CODE REVIEW

**Comment by DiegoVazquezNanini** [comment]
@michaelmcnamara Do we still need onward/conf/GeoIPCity-534-Location.csv?

File: common/app/views/support/cleaner/amp_embed_cleaner/AmpEmbed.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package views.support.cleaner.amp_embed_cleaner
```
Comment (jfsoul) [inline common/app/views/support/cleaner/amp_embed_cleaner/AmpEmbed.scala]: I would prefer `views.support.cleaner.amp.AmpEmbed`.  That way the other AMP cleaner can go in the `amp` package too (and we avoid the underscores in package names).

**Comment by michaelwmcnamara** [comment]
@DiegoVazquezNanini  I honestly don't know why that file is showing up as a change - I think it was a side-effect of merging in master. As far as I can tell the change is a deletion of the file.
I will seek advice tomorrow on how to extract the change to that filefrom my pull request.

**Comment by NataliaLKB** [comment]
@michaelwmcnamara, it seemed to happen in the commit: https://github.com/guardian/frontend/pull/16477/commits/153fc3aa176ec7059a02caf92f7cd2d923679c90.

Happy to sit with you and figure it out

**Comment by NataliaLKB** [comment]
@michaelwmcnamara I am closing this PR for now. Its been open too long and its pretty old. Please reopen when you are ready.

**Pull request closed without merging**